addons/l10n_it_edi/models/account_invoice.py

```diff
@@ -0,0 +1,757 @@
+            'subject': _('Sending file: %s') % (self.l10n_it_einvoice_id.name),
+            'body': _('Sending file: %s to ES: %s') % (self.l10n_it_einvoice_id.name, self.env.user.company_id.l10n_it_address_recipient_fatturapa),
+            'email_from': self.env.user.company_id.l10n_it_address_send_fatturapa,
+            'mail_server_id': self.env.user.company_id.l10n_it_mail_pec_server_id.id,
+            'attachment_ids': [(6, 0, self.l10n_it_einvoice_id.ids)],
+        })
+
+        mail_fattura = self.env['mail.mail'].create({
+            'mail_message_id': message.id,
+            'email_to': self.env.user.company_id.l10n_it_address_recipient_fatturapa,
+        })
+        try:
+            mail_fattura.send(raise_exception=True)
+            self.message_post(
+                body=(_("Mail sent on %s by %s") % (fields.Datetime.now(), self.env.user.display_name))
+                )
+            self.l10n_it_send_state = 'sent'
+        except MailDeliveryException as error:
+            self.message_post(
+                body=(_("Error when sending mail with E-Invoice: %s") % (error.args[0]))
+                )
+            self.l10n_it_send_state = 'invalid'
+
+    def _import_xml_invoice(self, content, attachment):
+        ''' Extract invoice values from the E-Invoice xml tree passed as parameter.
+
+        :param content: The tree of the xml file.
+        :return: A dictionary containing account.invoice values to create/update it.
+        '''
+
+        try:
+            tree = etree.fromstring(content)
+        except:
+            _logger.info('Error during decoding XML file')
+            return self.env['account.invoice']
+
+        invoices = self.env['account.invoice']
+
+        # possible to have multiple invoices in the case of an invoice batch, the batch itself is repeated for every invoice of the batch
+        for body_tree in tree.xpath('//FatturaElettronicaBody', namespaces=tree.nsmap):
+            self = self.with_context(default_journal_id=(self.env['account.journal'].search([('type', '=', 'purchase')], limit=1)).id)
+
+            elements = tree.xpath('//DatiGeneraliDocumento/TipoDocumento', namespaces=tree.nsmap)
+            if elements and elements[0].text and elements[0].text == 'TD01':
+                self_ctx = self.with_context(type='in_invoice')
+            elif elements and elements[0].text and elements[0].text == 'TD04':
+                self_ctx = self.with_context(type='in_refund')
+            else:
+                _logger.info(_('Document type not managed: %s.') % (elements[0].text))
+
+            # type must be present in the context to get the right behavior of the _default_journal method (account.invoice).
+            # journal_id must be present in the context to get the right behavior of the _default_account method (account.invoice.line).
+
+            journal_id = self_ctx._default_journal().id
+            self_ctx = self_ctx.with_context(journal_id=journal_id)
```

Comment:
should check multi-company -> in theory by checking in the xml to who you send it